src/widgetresize/resizer.js

```diff
@@ -187,30 +183,39 @@ export default class Resizer {
 
 	/**
 	 * Redraws the resizer.
+	 *
+	 * @param {module:utils/dom/rect~Rect} handleHostRect
 	 */
-	redraw() {
+	redraw( handleHostRect ) {
 		// TODO review this
 		const domWrapper = this._domResizerWrapper;
 
 		if ( existsInDom( domWrapper ) ) {
 			// Refresh only if resizer exists in the DOM.
 			const widgetWrapper = domWrapper.parentElement;
 			const handleHost = this._getHandleHost();
-			const clientRect = new Rect( handleHost );
+			const clientRect = handleHostRect || new Rect( handleHost );
 
 			domWrapper.style.width = clientRect.width + 'px';
 			domWrapper.style.height = clientRect.height + 'px';
 
+			const offsets = {
```

Comment:
It's meant to execute fetch all `HTMLElement.offsetX` subsequently, so that no unnecesarry layout thrashes are occurring.

Without this setting a `domWrapper.style.height` and then fetching `offsetX` forces layout recalculation, as the width might have changed as the height style was modified.

See this demo as an explainer https://codepen.io/mlewand/pen/zYOJENX

Optimized:

![image](https://user-images.githubusercontent.com/5353898/65121531-a7f87580-d9ef-11e9-9a73-ceb6bcc58a02.png)

vs

![image](https://user-images.githubusercontent.com/5353898/65127603-0e819180-d9f8-11e9-9431-56d2127802c7.png)

Reply:
But why an object and not 4 variables? This is extremely local code.